### ISSUE TITLE
Sub-optimal query with unnecessary not null

### ISSUE THREAD

**Issue by sitepodmatt** [dotnet/efcore#18239]
```   SELECT "a"."Id"
      FROM "Animals" AS "a"
      WHERE ((
          SELECT "t"."Value"
          FROM "text_attributes" AS "t"
          WHERE ("a"."Id" = "t"."AnimalId") AND (("t"."Name" = 'placeOfBirth') AND "t"."Name" IS NOT NULL)
          LIMIT 1) = 'seattle') AND (
          SELECT "t"."Value"
          FROM "text_attributes" AS "t"
          WHERE ("a"."Id" = "t"."AnimalId") AND (("t"."Name" = 'placeOfBirth') AND "t"."Name" IS NOT NULL)
          LIMIT 1) IS NOT NULL

```
As far as understand the not null checks aren't required in the above scenario, both in the predicate on testing Name, but also the whole redundant duplicate SQL statement.  I understand the issue with database nulls, and if we were doing "not equals" and we would need to consider null values too, but when we are testing equality against a constant this overzealous approach seems inefficient. Is the query builder too generalized, this seems like a low hanging fruit win for performance? 

The above is the result of:
```
   var results = ctx.Animals
                    .Where(a => a.TextAttributes.FirstOrDefault(x => x.Name == "placeOfBirth").Value == "seattle").ToList()
````
 
Where Animals has a List<TextAttribute> where TextAttribute has Name/Value string properties. 

tl;dr - not null checks aren't required

**Comment by ajcvickers** [comment]
@maumar to look for dupe

**Comment by ajcvickers** [comment]
@maumar @smitpatel to find dupe.

**Comment by smitpatel** [comment]
Duplicate of #17543

**Comment by ajcvickers** [comment]
@smitpatel Why did you remove `customer-reported` from this issue?

**Comment by smitpatel** [comment]
By mistake when removing type-bug.
